modules/rtdModule/index.js

```diff
@@ -169,8 +169,43 @@ let _userConsent;
  */
 export function attachRealTimeDataProvider(submodule) {
   registeredSubModules.push(submodule);
+  return function detach() {
+    const idx = registeredSubModules.indexOf(submodule)
+    if (idx >= 0) {
+      registeredSubModules.splice(idx, 1);
+      initSubModules();
+    }
+  }
```

Comment:
I'm not sure I understand why we need to return the detach function and call initSubmodules func again? I understand that it's getting call in the unit tests file, but does it make sense to have it here as a return value of `attachRealTimeDataProvider` function. What are your thoughts?

Reply:
I think this is the best way to do it, but I'll take suggestions. Do you prefer a separate global "deinit" function, or something else?
(I think this is the way because an undocumented return value that noone is using at the moment is less ugly than a public "resetSubmodules" function that is only called by the tests. In fact, the real way to do it would be to de-globalize the whole module, but I didn't feel like committing to a huge refactor for this issue).

Reply:
Sure, I somewhat agree with what you're saying. I don't have a strict preference for either of the approaches here, between yours and the global func 'resetSubmodules'. My concern was the func name, does it represent what is being implemented inside. Would it be possible to at least update the description of the function and document the return value.

Reply:
You're right - updated the docs.